src/app/shared/utils/browser-only.pipe.ts

```diff
@@ -0,0 +1,24 @@
+import { Inject, Pipe, PipeTransform, PLATFORM_ID } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+
+/**
+ * A pipe that only returns its intput when run in the browser.
+ * Used to distinguish client-side rendered content from server-side rendered content.
+ */
+@Pipe({
+  name: 'dsBrowserOnly'
+})
+export class BrowserOnlyPipe implements PipeTransform {
```

Comment:
First off, glad you figured out a way to work around the e2e test issues!

Could we provide a more detailed description of this new `dsBrowserOnly` pipe?  Correct me if I'm wrong, but it appears this might now be needed for **every component/button** you want to interact with in e2e tests?  Or is it somehow only needed for some e2e tests?

Based on your recent changes, it appears to me that this might be needed anywhere we add a `data-test` attribute, like:
```
[attr.data-test]="'value' | dsBrowserOnly"
```
But, I'm not sure if I'm overlooking something.  I also saw in the code commit that you mentioned this is used to essentially "block" e2e tests from running until Client side rendering (CSR) kicks in.  

Overall, I think a detailed description is warranted. You've got a good start of a description here, but it could be improved with details of how to use this for e2e tests & why it's necessary.

Reply:
On it!

> Correct me if I'm wrong, but it appears this might now be needed for every component/button you want to interact with in e2e tests? Or is it somehow only needed for some e2e tests?

I don't think we need `dsBrowserOnly` on _every_ button/input;
Only about 5 suites were failing in the first place, e.g. `submission.spec.ts` contains a bunch of interactions that were completely fine apparently, to be honest I don't know what makes the difference.

> I also saw in the code commit that you mentioned this is used to essentially "block" e2e tests from running until Client side rendering (CSR) kicks in. Is that accurate?

Yes.
The pipe ensures that SSR HTML won't contain the attribute, so Cypress will wait until CSR is ready for that particular component before interacting with it.

Reply:
>The pipe ensures that SSR HTML won't contain the attribute, so Cypress will wait until CSR is ready for that particular component before interacting with it.

To elaborate on that: those particular tests interact with components that need JS to function, so they wouldn't work on the pure SSR HTML. That's why we needed a way to differentiate between the SSR and the CSR version to know when to perform the test